### PR TITLE
SUBMARINE-1132. Upgrade model UI and complete fetching functionality

### PR DESCRIPTION
**Pull request by andybi7676**
### What is this PR for?

Upgrade model UI pages and complete routing & fetching functionality.

### What type of PR is it?
Improvement | Feature

### Todos
* [x] - Fetching functionality in model information page.
* [x] - Enable routing by breadcrumbs. 
* [x] - Improve UI looking. 
* [x] - Loading icon in every page. 
* [x] - Reorganize / Rename some components. 

### What is the Jira issue?

https://issues.apache.org/jira/browse/SUBMARINE-1132
### How should this be tested?

See screenshots.
### Screenshots (if appropriate)

https://user-images.githubusercontent.com/59821107/146330254-e581903b-cb5a-483a-b246-807c6bd0a1b3.mp4

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by andybi7676** [comment]
@KUAN-HSUN-LI I added the record and it is now ready for review, thanks!

**Comment by andybi7676** [comment]
And the tag filtering functionality in the registered-model info page has been excluded from this PR because I think the model tag component needs to be refactored, and I'll discuss it with you in the next group meeting.

File: submarine-workbench/workbench-web/src/app/pages/workbench/model/model-info/model-info.component.html [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      </tr>
+    </thead>
+    <tbody>
+      <tr *ngFor="let data of basicTable.data; let i = index">
+        <td><a [routerLink]="data.version">{{ data.version }}</a></td>
+        <td>{{ data.currentStage }}</td>
+        <td>
+          <submarine-model-tag *ngFor="let tag of data.tags" [tag]="tag" [cssType]="'default'"></submarine-model-tag>
+        </td>
+        <td>{{ data.description }}</td>
```
Comment (KUAN-HSUN-LI) [inline submarine-workbench/workbench-web/src/app/pages/workbench/model/model-info/model-info.component.html:74]: Is it better to move the tags and description to the last two columns?

File: submarine-workbench/workbench-web/src/app/pages/workbench/model/model-version/model-version.component.html [hunk 2]
```diff
@@ -16,20 +16,56 @@
...
+      <p nz-typography>
+        <span><strong>Description: </strong></span>
+        {{ isLoading ? null : modelVersionInfo.description }}
+      </p>
+      <p nz-typography>
+        <span><strong>Tags: </strong></span>
+        <span  *ngIf="!isLoading">
+          <submarine-model-tag *ngFor="let tag of modelVersionInfo.tags" [tag]="tag" [cssType]="'default'"></submarine-model-tag>
+        </span>
+      </p>
```
Comment (KUAN-HSUN-LI) [inline submarine-workbench/workbench-web/src/app/pages/workbench/model/model-version/model-version.component.html:32]: Same as above.

File: submarine-workbench/workbench-web/src/app/pages/workbench/model/model-version/model-version.component.html [hunk 3]
```diff
@@ -16,20 +16,56 @@
...
+      <p nz-typography>
+        <span><strong>Stage: </strong></span>
+        {{ isLoading ? null : modelVersionInfo.currentStage }}
+      </p>
+      <nz-divider [nzText]="text" nzOrientation="left">
+        <ng-template #text>
+          Model Version Info
+          <i nz-icon nzType="info-circle"></i>
+        </ng-template>
```
Comment (KUAN-HSUN-LI) [inline submarine-workbench/workbench-web/src/app/pages/workbench/model/model-version/model-version.component.html:41]: Can you explain why adding this line?
I think all of the metadata is model version info.

File: submarine-workbench/workbench-web/src/app/pages/workbench/model/model-info/model-info.component.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      <p nz-typography>
+        <span><strong>Description: </strong></span>
+        {{ isModelInfoLoading ? null : selectedModelInfo.description }}
+      </p>
+      <p nz-typography>
+        <span><strong>Tags: </strong></span>
+        <span *ngIf="!isModelInfoLoading">
+          <submarine-model-tag *ngFor="let tag of selectedModelInfo.tags" [tag]="tag" [cssType]="'default'"></submarine-model-tag>
+        </span>
+      </p>
```
Comment (KUAN-HSUN-LI) [inline submarine-workbench/workbench-web/src/app/pages/workbench/model/model-info/model-info.component.html:32]: Is it better to move the tags and description to the last two rows?

**Review by KUAN-HSUN-LI — COMMENTED** [review]
There are some discussions. Others LGTM.

**Review by KUAN-HSUN-LI — APPROVED** [review]
LGTM

**Pull request closed without merging**